src/stats.rs

```diff
@@ -403,21 +453,41 @@ impl Collector {
             // Some are counters, some are gauges...
             match stat.name {
                 EventName::Query => {
+                    let mut guard = CLIENT_STATES.write();
```

Comment:
You don't need a mutex, the stats collector is not multi-threaded.

Reply:
This lock here is needed because this hashmap is also read by admin.rs to construct `SHOW CLIENTS`

Access pattern for this data structure is mostly writes (updating client stats), reads are only triggered by `SHOW CLIENTS` so my strategy here is to acquire the read lock enough to clone the hashmap, unlock and then use it. I am open for suggestions though

Reply:
Checkout how we do it for regular stats. The lock exists but it's only used by the collect() loop. Clients use channels instead which I believe to be faster because they should be lock-free.